release/branch/cloudbuild.yaml

```diff
@@ -86,60 +86,60 @@ steps:
 
   # remove any dirty files after running the build
   - name: gcr.io/cloud-builders/git
-    dir: 'kpt'
+    dir: 'go/src/github.com/GoogleContainerTools/kpt'
     args: ['reset', '--hard']
     id: 'git-reset'
 
   # FYI: If cross-platform build issues happen, then stop caching the modules in the volume.
   # build windows
   - name: gcr.io/cloud-builders/go
-    env: ['GOOS=windows', 'GOARCH=amd64', 'CGO_ENABLED=0', 'GO111MODULE=on']
-    args: ['go', 'build', '-ldflags', '-X main.version=${BRANCH_NAME}', '-o', '/workspace/bin/${BRANCH_NAME}/windows_amd64/kpt.exe', '.']
-    dir: 'kpt'
+    env: ['GOPATH=/workspace/go', 'GOOS=windows', 'GOARCH=amd64', 'CGO_ENABLED=0', 'GO111MODULE=on']
+    args: ['build', '-ldflags', '-X main.version=master', '-o', '/workspace/bin/master/windows_amd64/kpt.exe', '.']
+    dir: 'go/src/github.com/GoogleContainerTools/kpt'
     volumes:
       - name: go-modules
         path: /go
     waitFor: ['git-reset']
 
   # build linux
   - name: gcr.io/cloud-builders/go
-    env: ['GOOS=linux', 'GOARCH=amd64', 'CGO_ENABLED=0', 'GO111MODULE=on']
-    args: ['go', 'build', '-ldflags', '-X main.version=${BRANCH_NAME}', '-o', '/workspace/bin/${BRANCH_NAME}/linux_amd64/kpt', '.']
-    dir: 'kpt'
+    env: ['GOPATH=/workspace/go', 'GOOS=linux', 'GOARCH=amd64', 'CGO_ENABLED=0', 'GO111MODULE=on']
+    args: ['build', '-ldflags', '-X main.version=master', '-o', '/workspace/bin/master/linux_amd64/kpt', '.']
+    dir: 'go/src/github.com/GoogleContainerTools/kpt'
     volumes:
       - name: go-modules
         path: /go
     waitFor: ['git-reset']
 
   # build darwin
   - name: gcr.io/cloud-builders/go
-    env: ['GOOS=darwin', 'GOARCH=amd64', 'CGO_ENABLED=0', 'GO111MODULE=on']
-    args: ['go', 'build', '-ldflags', '-X main.version=${BRANCH_NAME}', '-o', '/workspace/bin/${BRANCH_NAME}/darwin_amd64/kpt', '.']
-    dir: 'kpt'
+    env: ['GOPATH=/workspace/go', 'GOOS=darwin', 'GOARCH=amd64', 'CGO_ENABLED=0', 'GO111MODULE=on']
+    args: ['build', '-ldflags', '-X main.version=master', '-o', '/workspace/bin/master/darwin_amd64/kpt', '.']
+    dir: 'go/src/github.com/GoogleContainerTools/kpt'
     volumes:
       - name: go-modules
         path: /go
     waitFor: ['git-reset']
 
   # build docker image
   - name: 'gcr.io/cloud-builders/docker'
-    args: [ 'build', '-t', 'gcr.io/$PROJECT_ID/kpt:${BRANCH_NAME}', '-f', 'release/Dockerfile', '.' ]
-    dir: 'kpt'
+    args: [ 'build', '-t', 'gcr.io/$PROJECT_ID/kpt:master', '-f', 'release/Dockerfile', '.' ]
+    dir: 'go/src/github.com/GoogleContainerTools/kpt'
 
   # push the container image
   - name: 'gcr.io/cloud-builders/docker'
-    args: ['push', 'gcr.io/$PROJECT_ID/kpt:${BRANCH_NAME}']
-    dir: 'kpt'
+    args: ['push', 'gcr.io/$PROJECT_ID/kpt:master']
+    dir: 'go/src/github.com/GoogleContainerTools/kpt'
 
   # push the binaries
   - name: 'gcr.io/cloud-builders/gsutil'
     args: ['cp', '-r', '-a', 'public-read', '/workspace/bin/', 'gs://${_GCS_BUCKET}/']
```

Comment:
Not much familiar with the setup, so take it with a grain of salt. If I understand correctly, the binaries for `v0` branch will be available under gsbucket/.../master/kpt_OS_ARCH, right ? if that is correct, then, it can get confusing for people because people think `master` has the latest. Not sure, how much of a big deal that is.

Reply:
The files are pushed to GCS under `kpt-dev/bin/master/OS_ARCH/kpt(.exe)`. We also publish docker images as `gcr.io/kpt-dev/kpt:master`. Users might rely on these stable URIs for fetching the latest (pre 1.0) version of kpt. Changing from master to v0 would break this, so I think we should just keep using master. Relatively soon we will discontinue support for that branch and we can remove the old binaries.

Reply:
Got it. Thanks.